modules/nomad-vault-integration/main.tf

```diff
@@ -0,0 +1,89 @@
+#################################################
+# AWS Authentication
+#################################################
+resource "vault_auth_backend" "aws" {
+    type = "aws"
+    path = "${var.aws_auth_path}"
+}
+
+################################################
+# Nomad Server Policy
+# See https://www.nomadproject.io/docs/vault-integration/index.html
+################################################
+data "template_file" "nomad_server_policy" {
+    template = "${file("${path.module}/templates/nomad_server_policy.hcl")}"
+
+    vars {
+        nomad_token_role = "${var.nomad_token_role}"
+    }
+}
+
+resource "vault_policy" "nomad_server_policy" {
+    name = "${var.nomad_server_policy}"
+
+    policy = "${data.template_file.nomad_server_policy.rendered}"
+}
+
+################################################
+# AWS Authentication Token Role
+################################################
+data "template_file" "nomad_aws_token_role" {
+    template = "${file("${path.module}/templates/vault_aws_role.json")}"
+
+    vars {
+        nomad_server_policy = "${var.nomad_server_policy}"
+        nomad_token_role = "${var.nomad_token_role}"
```

Comment:
Current `vault_aws_role.json` doesn't seem to use `nomad_token_role`, so the variable is not required.